src/UserService.Business/Commands/Avatar/RemoveAvatarsCommand.cs

```diff
@@ -0,0 +1,140 @@
+{
+  public class RemoveAvatarsCommand : IRemoveAvatarsCommand
+  {
+    private readonly IAvatarRepository _avatarRepository;
+    private readonly IUserRepository _userRepository;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IRequestClient<IRemoveImagesRequest> _rcRemoveImages;
+    private readonly IRemoveAvatarsRequestValidator _removeRequestValidator;
+    private readonly IAccessValidator _accessValidator;
+    private readonly ILogger<RemoveAvatarsCommand> _logger;
+
+    private bool RemoveImages(List<Guid> imageIds, Guid userId, List<string> errors)
+    {
+      bool result = false;
+
+      string errorMessage = "Can't remove images. Please try again later.";
+      string logMessage = "Errors while removing images with ids: {Ids}.";
+
+      try
+      {
+        Response<IOperationResult<bool>> removeResponse = _rcRemoveImages.GetResponse<IOperationResult<bool>>(
+          IRemoveImagesRequest.CreateObj(imageIds, ImageSource.User))
+          .Result;
+
+        if (removeResponse.Message.IsSuccess)
+        {
+          _avatarRepository.Remove(imageIds);
+
+          DbUser dbUser = _userRepository.Get(userId);
+
+          if (dbUser.AvatarFileId != null && imageIds.Contains(dbUser.AvatarFileId.Value))
+          {
+            _userRepository.UpdateAvatar(userId, null);
+          }
+
+          result = removeResponse.Message.Body;
+        }
+        else
+        {
+          string warningMessage = logMessage + "Errors: {Errors}";
+
+          _logger.LogWarning(
+            warningMessage,
+            string.Join(", ", imageIds),
+            string.Join('\n', removeResponse.Message.Errors));
+
+          errors.Add(errorMessage);
+        }
+      }
+      catch (Exception e)
+      {
+        _logger.LogError(e, logMessage, string.Join(", ", imageIds));
+
+        errors.Add(errorMessage);
+      }
+
+      return result;
+    }
+
+    public RemoveAvatarsCommand(
+      IAvatarRepository avatarRepository,
+      IUserRepository userRepository,
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IRemoveImagesRequest> rcRemoveImages,
+      IRemoveAvatarsRequestValidator removeRequestValidator,
+      IAccessValidator accessValidator,
+      ILogger<RemoveAvatarsCommand> logger)
+    {
+      _avatarRepository = avatarRepository;
+      _userRepository = userRepository;
+      _httpContextAccessor = httpContextAccessor;
+      _rcRemoveImages = rcRemoveImages;
+      _removeRequestValidator = removeRequestValidator;
+      _accessValidator = accessValidator;
+      _logger = logger;
+    }
+
+    public OperationResultResponse<bool> Execute(RemoveAvatarsRequest request)
+    {
+      OperationResultResponse<bool> response = new();
+      List<string> errors = new();
+
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser dbUser = _userRepository.Get(senderId);
+
+      if (!(dbUser.IsAdmin ||
```

Comment:
update kernel if you don't use last version and use only HasRights. Has rights contain 'check admin' request now